Polyhedron/demo/Polyhedron/Scene.cpp

```diff
@@ -432,83 +446,116 @@ glDepthFunc(GL_LEQUAL);
+                if (depth != 1.0)
+                {
+                    //add object to list of picked objects;
+                    picked_item_IDs[depth] = index;
+                }
+            }
         }
     }
-    glDepthFunc(GL_LESS);
-    // Splatting
-    if(!with_names && ms_splatting->isSupported())
+    if(!with_names)
     {
-
-        ms_splatting->beginVisibilityPass();
-        for(int index = 0; index < m_entries.size(); ++index)
+        glDepthFunc(GL_LESS);
+        // Splatting
+        if(!with_names && ms_splatting->isSupported())
         {
-            CGAL::Three::Scene_item& item = *m_entries[index];
-            if(item.visible() && item.renderingMode() == Splatting)
+
+            ms_splatting->beginVisibilityPass();
+            for(int index = 0; index < m_entries.size(); ++index)
             {
+                CGAL::Three::Scene_item& item = *m_entries[index];
+                if(item.visible() && item.renderingMode() == Splatting)
+                {
 
-                if(viewer)
+                    if(viewer)
+                    {
+                        item.draw_splats(viewer);
+                    }
+                    else
+                        item.draw_splats();
+                }
+
+            }
+            ms_splatting->beginAttributePass();
+            for(int index = 0; index < m_entries.size(); ++index)
+            {  CGAL::Three::Scene_item& item = *m_entries[index];
+                if(item.visible() && item.renderingMode() == Splatting)
                 {
-                    item.draw_splats(viewer);
+                    viewer->glColor4d(item.color().redF(), item.color().greenF(), item.color().blueF(), item.color().alphaF());
+                    if(viewer)
+                        item.draw_splats(viewer);
+                    else
+                        item.draw_splats();
                 }
-                else
-                    item.draw_splats();
             }
+            ms_splatting->finalize();
 
         }
-       ms_splatting->beginAttributePass();
-         for(int index = 0; index < m_entries.size(); ++index)
-        {  CGAL::Three::Scene_item& item = *m_entries[index];
-            if(item.visible() && item.renderingMode() == Splatting)
-            {
-                viewer->glColor4d(item.color().redF(), item.color().greenF(), item.color().blueF(), item.color().alphaF());
-                if(viewer)
-                    item.draw_splats(viewer);
-                else
-                    item.draw_splats();
-            }
-        }
-        ms_splatting->finalize();
-
     }
+    else
+    {
+        QList<float> depths = picked_item_IDs.keys();
+        if(!depths.isEmpty())
+        {
+            qSort(depths);
+            int id = picked_item_IDs[depths.first()];
+            setSelectedItemIndex(id);
+            viewer->setSelectedName(id);
 
-    //scrolls the sceneView to the selected item's line.
```

Comment:
A `QMap` like `picked_item_IDs` is always sorted by keys, unless a `QHashMap`. No need for `QList<float> depths`:
```C++
if(!picked_item_IDs.isEmpty()) {
   int id = picked_item_IDs.first();
   // [...]
}
```